osu.Game/Beatmaps/BeatmapDifficultyManager.cs

```diff
@@ -300,11 +301,13 @@ public BindableStarDifficulty(BeatmapInfo beatmap, CancellationToken cancellatio
         public readonly double Stars;
         public readonly int MaxCombo;
 
-        public StarDifficulty(double stars, int maxCombo)
+        public readonly DifficultyAttributes Attributes;
+
+        public StarDifficulty(double stars, int maxCombo, DifficultyAttributes attributes = null)
```

Comment:
This nullability of `attributes` worries me. Why can't it be guaranteed to be present?

Reply:
Also may be best to wait for score standardisation PRs before this, because they add MaxCombo to `DifficultyAttributes`, so the extra params here would only be duplication.

Reply:
See this line as to why the difficulty attributes can't be guaranted to be present: https://github.com/Game4all/osu/blob/2766cf73b492ea751723b387399a40c4b6e20bc3/osu.Game/Beatmaps/BeatmapDifficultyManager.cs#L237